MovieQuiz/Presentation/ AlertPresenter.swift

```diff
@@ -0,0 +1,33 @@
+//
+//   AlertPresenter.swift
+//  MovieQuiz
+//
+//  Created by Alexander Farizanov on 04.12.2022.
+//
+
+import Foundation
+import UIKit
+
+class AlertPresenter {
+    
+    private var questionFactory: QuestionFactoryProtocol? = nil
+    weak var alertController : UIViewController?
+    init(alertController: UIViewController?)
+    {
+        self.alertController = alertController
+    }
+    
+    func show(result : AlertModel) {
+        let alert = UIAlertController(title: result.title, message: result.message, preferredStyle: .alert)
+        let action = UIAlertAction(title: result.buttonText, style: .default)
+      //      guard let completion? = result.completion else { return }
+       //                    completion()
+     
+        alert.addAction(action)
+        
+        alertController?.present(alert, animated: true, completion: nil)
+        UIView.animate(withDuration: 0, delay: 1.5, options: [], animations: { })
+            //questionFactory?.requestNextQuestion()
+        
+    }
```

Comment:
> Обработка нажатия на кнопку алерт реализована с использованием либо замыкания, либо делегата.

В AlertModel складываем в completion блок кода, который хотим выполнить по нажатию на кнопку. И создаем UIAlertAction с третьим параметром - замыканием, где этот блок кода вызываем: 
```suggestion
    func show(result : AlertModel) {
        let alert = UIAlertController(title: result.title, message: result.message, preferredStyle: .alert)
        let action = UIAlertAction(title: result.buttonText, style: .default) { 
            result.completion()
        }
        alert.addAction(action)
        
        viewController?.present(alert, animated: true, completion: nil)// просила переименовать в контроллер
    }
```

лишнее удалила, как просила в предыдущем ревью

Reply:
Я в итоге запихал его в handler UIAlertAction

`let action = UIAlertAction(title: result.buttonText, style: .default, handler: result.completion)`

Почему стилисчически правильно именно переименовать? Не совсем понятно - так же название больше говорит за себя?